requirements.txt

```diff
@@ -18,6 +18,7 @@ trio==0.24.0
 coverage==7.4.1
 types-PyYAML==6.0.12.12
 moto[s3]==4.2.14
+respx==0.20.2
```

Comment:
Let's avoid introducing a new dependency. Instead, we can use mocking classes that hishel provides. See `MockTransport` and `MockAsyncTransport`